a10_octavia/controller/worker/tasks/nat_pool_tasks.py

```diff
@@ -29,24 +29,49 @@ class NatPoolCreate(task.Task):
     def execute(self, loadbalancer, vthunder, flavor_data=None):
         device_pool = None
         if flavor_data:
+            natpool_flavor_list = flavor_data.get('nat_pool_list')
             natpool_flavor = flavor_data.get('nat_pool')
+            if natpool_flavor_list:
+                for i in range(len(natpool_flavor_list)):
+                    if self.axapi_client.nat.pool.exists(natpool_flavor_list[i]['pool_name']):
+                        device_pool = self.axapi_client.nat.pool.get(
+                            natpool_flavor_list[i]['pool_name'])
+                    if not (device_pool and
+                            (device_pool['pool']['start-address'] ==
+                                natpool_flavor_list[i]['start_address'] and
+                                device_pool['pool']['end-address'] ==
+                                natpool_flavor_list[i]['end_address'])):
+                        pool_name = natpool_flavor_list[i]['pool_name']
+                        start_address = natpool_flavor_list[i]['start_address']
+                        end_address = natpool_flavor_list[i]['end_address']
+                        netmask = natpool_flavor_list[i]['netmask']
+                        gateway = natpool_flavor_list[i]['gateway']
+                        try:
+                            self.axapi_client.nat.pool.create(
+                                pool_name, start_address, end_address, netmask, gateway=gateway)
+                        except(acos_errors.Exists) as e:
+                            LOG.exception(
+                                "The specified nat-pool already exists on thunder")
```

Comment:
Though this is more of a defensive code section (given that we should never trigger this exception), it should still include the nat-pool name, flavor-id, lb-id. Ex:

"Nat-pool with name {} already exists on partition {} of thunder device {}"

Should this worst case occur, we still want an operator to know how to recover on their end.

Reply:
Changed the exception message.

Reply:
@hthompson-a10 
openstack didn't allow update (or set) flavorprofile once it is in use. So, this is fine.

`stack@ytsai-openstack:~/source/a10-octavia/snat$ openstack loadbalancer flavorprofile set --flavor-data '' fp_snat
Flavor profile 93bec7f8-9f39-44ee-bc75-1d0e8f8fe8f6 is in use and cannot be modified. (HTTP 409) (Request-ID: req-c5eb45d8-6bfe-4aeb-b869-a8d847ee4c92)
`